v2/blockstore/readwrite_test.go

```diff
@@ -945,6 +945,38 @@ func TestReadWrite_ReWritingCARv1WithIdentityCidIsIdenticalToOriginalWithOptions
 	require.Equal(t, wantSum, gotSum)
 }
 
+func TestReadWriteOpenFile(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	dir := t.TempDir() // auto cleanup
+	f, err := ioutil.TempFile(dir, "")
+	require.NoError(t, err)
+
+	root := blocks.NewBlock([]byte("foo"))
+
+	bs, err := blockstore.OpenReadWriteFile(f, []cid.Cid{root.Cid()})
+	require.NoError(t, err)
+
+	err = bs.Put(ctx, root)
+	require.NoError(t, err)
+
+	roots, err := bs.Roots()
+	require.NoError(t, err)
+	_, err = bs.Has(ctx, roots[0])
+	require.NoError(t, err)
+	_, err = bs.Get(ctx, roots[0])
+	require.NoError(t, err)
+	_, err = bs.GetSize(ctx, roots[0])
+	require.NoError(t, err)
+
+	err = bs.Finalize()
+	require.NoError(t, err)
+
+	_, err = f.Seek(0, 0)
+	require.NoError(t, err) // file should not be closed, let the caller do it
```

Comment:
The test is the caller, and therefore should close the file. No?

Reply:
at least windows seems to agree with you, I'll add that